# Project 4

In this project, you will summarize and present your analysis from Projects 1-3.

### Intro: Write a problem Statement/ Specific Aim for this project

Answer: We wish to determine how strongly undergraduate university prestige is correlated with graduate school admissions.

### Dataset:  Write up a description of your data and any cleaning that was completed

Answer: Our data is a CSV consisting of an outcome (admission status), GRE scores, GPAs, and undergraduate prestige. There are 400 admission statuses, but not all rows are complete. We drop them rather than fill, because there are very few.

### Demo: Provide a table that explains the data by admission status

In [16]:
import pandas as pd
df = pd.read_csv("admissions.csv")
df = pd.get_dummies(df, columns=['prestige'], prefix='prestige')
# non-admissions
print df.loc[df['admit'] == 0].describe()
# admissions
print df.loc[df['admit'] == 1].describe()

       admit         gre         gpa  prestige_1.0  prestige_2.0  \
count  273.0  271.000000  272.000000    273.000000    273.000000   
mean     0.0  573.579336    3.345404      0.102564      0.355311   
std      0.0  116.052798    0.376773      0.303946      0.479487   
min      0.0  220.000000    2.260000      0.000000      0.000000   
25%      0.0  500.000000    3.080000      0.000000      0.000000   
50%      0.0  580.000000    3.340000      0.000000      0.000000   
75%      0.0  660.000000    3.610000      0.000000      1.000000   
max      0.0  800.000000    4.000000      1.000000      1.000000   

       prestige_3.0  prestige_4.0  
count    273.000000    273.000000  
mean       0.340659      0.201465  
std        0.474801      0.401831  
min        0.000000      0.000000  
25%        0.000000      0.000000  
50%        0.000000      0.000000  
75%        1.000000      0.000000  
max        1.000000      1.000000  
       admit         gre         gpa  prestige_1.0  prestige_2.

Mean (STD) or counts by admission status for each variable

| Not Admitted | Admitted
---| ---|---
GPA | 574 (116)  | 619 (109) 
GRE | 3.35 (0.37) | 3.49 (0.37)
Prestige 1 | 10.3% (%) | 26% (%)
Prestige 2 | 35.5% (%) | 41.7% (%)
Prestige 3 | 34.1% (%) | 22% (%)
Prestige 4 | 20.1% (%) | 9.4% (%)

### Methods: Write up the methods used in your analysis

Answer: To analyze this data, we choose a logistic regression to model our data. Our outcome is admission; our variables are prestige, GRE, and GPA. If the hypothesis is correct, prestige should have an outsize impact on admission.

### Results: Write up your results

Answer: Our hypothesis was correct: prestige has a larger impact on admission than GRE or GPA. The model predicted that someone who attended a prestige=1 school is twice as likely to be admitted to graduate school than someone from a prestige=2 school, holding all else fixed. We can see this from the coefficients of our regression; the coefficients of prestige=4 is -1.55, indicating that if someone attended a prestige=4 school, their admission odds drop preciptiously.

### Visuals: Provide a table or visualization of these results

<img src="gre_vs_predictions.png">

<img src='gpa_vs_predictions.png'>

### Discussion: Write up your discussion and future steps

Answer: As we can see from these two graphs, at every point, prestige=1 students have a higher chance of being admitted to graduate school. Furthermore, you can see from the general _shape_ of the lines (plus comparing them together) that GRE and GPA don't have an outsize impact vs prestige--that the shapes are similar, indeed, nearly identical, indicates that GRE and GPA follow the same pattern: they raise your chances of admission, but do not control it. Witness prestige=4 students with high GRE scores and GPAs, whose predicted probability of admission in either chart barely surpasses those of the prestige=1 students with low GREs and GPAs. 

A future direction could be to join this data set with other, relevant data sets: race, age, family income, etc. could all provide further insight into admissions. Furthermore, other models, such as decision trees, may prove fruitful as well.